test/virtual/include/openenclave/edger8r/common.h

```diff
@@ -66,35 +66,46 @@ OE_INLINE oe_result_t oe_add_size(size_t* total, size_t size)
     return result;
 }
 
-#define OE_ADD_SIZE(total, size)                                   \
-    do                                                             \
-    {                                                              \
-        if (sizeof(total) > sizeof(size_t) && total > OE_SIZE_MAX) \
-        {                                                          \
-            _result = OE_INVALID_PARAMETER;                        \
-            goto done;                                             \
-        }                                                          \
-        if (sizeof(size) > sizeof(size_t) && size > OE_SIZE_MAX)   \
-        {                                                          \
-            _result = OE_INVALID_PARAMETER;                        \
-            goto done;                                             \
-        }                                                          \
-        if (oe_add_size((size_t*)&total, (size_t)size) != OE_OK)   \
-        {                                                          \
-            _result = OE_INTEGER_OVERFLOW;                         \
-            goto done;                                             \
-        }                                                          \
+#define OE_MUL_SIZE(a, b, c)                         \
+    do                                               \
+    {                                                \
+        if ((a) && (b) && ((a) > OE_SIZE_MAX / (b))) \
+        {                                            \
+            _result = OE_INTEGER_OVERFLOW;           \
+            goto done;                               \
+        }                                            \
+        (c) = (a) * (b);                             \
+    } while (0)
+
+#define OE_ADD_SIZE(total, argcnt, argsize)                \
+    do                                                     \
+    {                                                      \
+        size_t __size = argsize;                           \
+        if (argcnt != 1)                                   \
+        {                                                  \
+            OE_MUL_SIZE(argcnt, argsize, __size);          \
+        }                                                  \
+        if (sizeof(total) != sizeof(size_t))               \
```

Comment:
This seems like a static assertion.Is there a better way to implement this rather than a runtime check?

Reply:
This check is from existing code.